src/osiris_log.erl

```diff
@@ -1884,6 +1821,61 @@ open_index_read(File) ->
     _ = file:read(Fd, ?IDX_HEADER_SIZE),
     Fd.
 
+scan_idx(IndexFile, Offset, LastChunkInSegment) when is_list(IndexFile) ->
+    IndexFd = open_index_read(IndexFile),
+    Result = scan_idx(IndexFd, Offset, LastChunkInSegment),
+    file:close(IndexFd),
+    Result;
+
+%% no known last chunk -> return the first chunk
+scan_idx(_Fd, _Offset, undefined) ->
+    {0, ?LOG_HEADER_SIZE};
```

Comment:
I know this is what the previous code did but it's not good. We don't have to fix it now though perhaps just add the old comment back